### PR TITLE
Increased resolution from ms to ns via hrtime and changed the interface

### PR DESCRIPTION
**Pull request by psotres**
I just coded some extra functionalities I needed that could be interesting to merge back to the base project. Hope you find them useful

- Time resolution is increased from ms to ns by using process.hrtime() instead of Date.now()
- Interface has been changed to allow Profiler instance retrieval by name from different scopes and files.
- New Profiler.profile() function to wrap methods (either sync or async) with Profiler.start() and Profiler.end() using the function name
- Changed Profiler.toString() signature to accept a new parameter for graph width
- Profiler.toString() relies on pretty-hrtime package to show durations using the most convinient unit

- Added new tests and increased code coverage to 100%

### CODE REVIEW

**Comment by gxcsoccer** [comment]
sorry for late reply

**Comment by gxcsoccer** [comment]
Looks great to me! I learned a lot from your pr @psotres 

But we need wait https://github.com/node-modules/time-profile/pull/5 to merge first

File: lib/profiler.js [hunk 1]
```diff
@@ -4,13 +4,29 @@ const assert = require('assert');
 const Base = require('sdk-base');
 const Entry = require('./entry');
 
+const _profilers = new Map();
+
+const isPromise = value =>
+  value !== null && typeof value === 'object' && typeof value.then === 'function';
```
Comment (gxcsoccer) [inline lib/profiler.js]:
```js
const is = require('is-type-of');

is.promise(p);
```

File: lib/profiler.js [hunk 2]
```diff
@@ -4,13 +4,29 @@ const assert = require('assert');
...
+  }
+
+  async profile(fn, ...args) {
+    this.start(fn.name);
+    let ret = fn(...args);
+    if (isPromise(ret)) {
+      ret = await ret;
+    }
+    this.end(fn.name);
+    return ret;
```
Comment (gxcsoccer) [inline lib/profiler.js]: 1. anonymous function 
```js
(() => {}).name // ''
```

2. duplication

```js
async function a() {
  await sleep(100);
}

profiler.profile(a);
profiler.profile(a);
```

File: lib/profiler.js (hunk 1 above)
Comment (psotres) [reply lib/profiler.js]: I would avoid an extra dependencies for a single simple function

File: lib/profiler.js (hunk 2 above)
Comment (psotres) [reply lib/profiler.js]: You are right but I like the current clean interface as is. An extra param would imply a null value in the middle of the call when fn.name is going to be used and function parameters are provided (~99% of the times).

What about an extra profileTagged(fn, tag, ...args) function?

**Comment by psotres** [comment]
Async/await is only available from node 8, so only versions after that one are compatible. Feel free to use old grammar based on generators if you prefer so.

**Comment by gxcsoccer** [comment]
+1

time-profile@2.0.0 is released, and I invited you @psotres to join node-modules group.

**Pull request merged**